code/create/dta_dicom2spec_rules.py

```diff
@@ -215,10 +160,36 @@ def _rules(self, series_dict, subject=None, anon_subject=None,
                 'bids-task': apply_bids_label_restrictions(_guess_task(self, series_dict) ),
                 'id': series_dict.get('SeriesNumber', 'unknown'),
 		'bids-run' : apply_bids_label_restrictions( run )
-              #  'bids-run' : apply_bids_label_restrictions(_guess_run(self, series_dict)) if "ProtocolName" in series_dict else None
                 }
 
+    # The table contains rows like "DTA003   T12122   T15722"
+    # so if you find a T-value in the table, you know, that the
+    # first field of that row contains the DTA003-value, that
+    # I have to use as the subject-ID
+    def get_column( self, t_value ):
+	# TODO: check if the relative path with .. in the beginning works.
+#         table_path = "/data/BnB1/DATA/source_data/DTA/code/dtage.tsv"
+         table_path = "../dtage.tsv"
+         with open( table_path, 'r' ) as table:
+            for line in table:
+               for part in line.split("\t"):
+                  if t_value in part:
+                     return str(line.split("\t").index(part))
+         return None
+
+
+    # this function reads a table and searches the given t-value (these are the session values for dta)
+    # when it finds the session value, it retunrs the matching subject-ID from the table.
+    def read_subject_for_session_from_table( self, t_value ):
+        table_path = "/data/BnB1/DATA/source_data/DTA/code/dtage.tsv"
+        with open( table_path, 'r' ) as table:
+            for line in table:
+                if t_value in line:
+                   return line.replace("\n","\t").split("\t")[0]
+
 
+    # TODO: maybe I sould use this function or build something equaly. Or delete it, if I am more sure about it.
+    # anyways, ExamCard is not a protocol-name of DTA at all.
```

Comment:
For the record: This is scanner or manufacturer specific. It's technical information rather than an actual imageseries. That's why it's considered invalid. You might find something similar in DTA, but depends on whether those DICOMs were published inn the first place.